xonsh/environ.py

```diff
@@ -482,7 +482,7 @@ def fromdircolors(cls, filename=None):
             out = subprocess.check_output(
                 cmd, env=denv, universal_newlines=True, stderr=subprocess.DEVNULL
             )
-        except (subprocess.CalledProcessError, FileNotFoundError):
+        except (subprocess.CalledProcessError, FileNotFoundError, OSError):
```

Comment:
`FileNotFoundError` is a child of `OSError` so having them both here is redundant, BUT, since this is a Windows-specific behavior, I'd prefer if we didn't swallow `OSError`s on non windows systems, so maybe we can have the `except OSError` in a separate clause and then raise the error if we aren't `ON_WINDOWS`?

something like

```python
except (subprocess.CalledProcessError, FileNotFoundError):
    return cls(cls.default_settings)
except OSError as e:
    if ON_WINDOWS:
       return cls(cls.default_settings)
    raise
```